apps/web/src/components/tx-flow/flows/SignMessageOnChain/index.tsx

```diff
@@ -1,40 +1,41 @@
-import TxLayout from '@/components/tx-flow/common/TxLayout'
-import type { TxStep } from '../../common/TxLayout'
 import { AppTitle } from '@/components/tx-flow/flows/SignMessage'
 import ReviewSignMessageOnChain, {
   type SignMessageOnChainProps,
 } from '@/components/tx-flow/flows/SignMessageOnChain/ReviewSignMessageOnChain'
-import { useMemo } from 'react'
-import useTxStepper from '../../useTxStepper'
-import { ConfirmSignMessageOnChainDetails } from './ConfirmSignMessageOnChainDetails'
+import { useCallback } from 'react'
 import { TxFlowType } from '@/services/analytics'
+import { type ReviewTransactionProps } from '@/components/tx/ReviewTransactionV2'
+import { type SubmitCallback, TxFlow } from '../../TxFlow'
+import { dispatchSafeAppsTx } from '@/services/tx/tx-sender'
 
 const SignMessageOnChainFlow = ({ props }: { props: Omit<SignMessageOnChainProps, 'onSubmit'> }) => {
-  const { step, nextStep, prevStep } = useTxStepper(undefined, TxFlowType.SIGN_MESSAGE_ON_CHAIN)
+  const { requestId } = props
+  const ReviewComponent = useCallback(
+    (reviewTxProps: ReviewTransactionProps) => {
+      return <ReviewSignMessageOnChain {...props} {...reviewTxProps} />
+    },
+    [props],
+  )
 
-  const steps = useMemo<TxStep[]>(
-    () => [
-      {
-        txLayoutProps: { title: 'Confirm message' },
-        content: <ReviewSignMessageOnChain {...props} key={0} onSubmit={() => nextStep(undefined)} />,
-      },
-      {
-        txLayoutProps: { title: 'Confirm message details', fixedNonce: true },
-        content: <ConfirmSignMessageOnChainDetails requestId={props.requestId} key={1} />,
-      },
-    ],
-    [nextStep, props],
+  const handleSubmit: SubmitCallback = useCallback(
+    async (args) => {
+      debugger
+      if (!args?.txId) {
+        return
+      }
+      const safeTxHash = args.txId.slice(-66)
```

Comment:
Nit: we should create a helper for this as I see it's done [in your other PR](https://github.com/safe-global/safe-wallet-monorepo/pull/5833/files#diff-a7a768266564667c635b0e58330eadee2f1369131743e74ae5554a779afb2002R39). We could also do this within `dispatchSafeAppsTx`.